### PR TITLE
VIH-6134 check audio file before delete

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-6134

### Change description ###
Check if audio file exists before delete audio app

**Does this PR introduce a breaking change?** (check one with "x")

```
[X ] Yes
[ ] No
```

### CODE REVIEW

File: VideoAPI/Video.API/Controllers/AudioRecordingController.cs [hunk 1]
```diff
@@ -214,19 +222,30 @@ public async Task<IActionResult> DeleteAudioStreamAsync(Guid hearingId)
...
                 return NotFound();
             }
-
+            
             var audioFileLink = await _storageService.CreateSharedAccessSignature(filePath, TimeSpan.FromDays(14));
-
             return Ok(new AudioRecordingResponse {AudioFileLink = audioFileLink});
         }
+
+         private async Task<bool> CheckAudioRecordingFile(Guid hearingId)
```
Comment (rehmanab) [inline VideoAPI/Video.API/Controllers/AudioRecordingController.cs:237]: If you change this to EnsureAudioFileExists and that THROWS an audioplatformexception, that would be better.

File: VideoAPI/Video.API/Controllers/AudioRecordingController.cs [hunk 2]
```diff
@@ -107,15 +108,22 @@ public async Task<IActionResult> CreateAudioApplicationWithStreamAsync(Guid hear
...
 
-            return NoContent();
+                if (!response.Success)
+                {
+                    return StatusCode((int)response.StatusCode, response.Message);
+                }
+
+                return NoContent();
+            }
+            else
```
Comment (rehmanab) [inline VideoAPI/Video.API/Controllers/AudioRecordingController.cs:123]: You dont need an else statement here

File: VideoAPI/Video.API/Controllers/ConferenceController.cs [hunk 3]
```diff
@@ -390,31 +393,46 @@ private async Task DeleteAudioRecordingApplication(Guid conferenceId)
...
             {
-                await _audioPlatformService.DeleteAudioApplicationAsync(queriedConference.HearingRefId);
+                if (await CheckAudioRecordingFile(queriedConference.HearingRefId))
+                {
+                    await _audioPlatformService.DeleteAudioApplicationAsync(queriedConference.HearingRefId);
+                }
+                else
+                {
+                    var msg = $"Audio recording file not found for hearing: {queriedConference.HearingRefId}";
+                    var ex = new AudioPlatformException(msg, HttpStatusCode.NotFound);
```
Comment (rehmanab) [inline VideoAPI/Video.API/Controllers/ConferenceController.cs:403]: Same here, should not be just creating exceptions, just for logging.. if you know what I mean..

File: VideoAPI/Video.API/Controllers/AudioRecordingController.cs (hunk 2 above)
Comment (valentinabennett) [reply VideoAPI/Video.API/Controllers/AudioRecordingController.cs:123]: done

File: VideoAPI/Video.API/Controllers/AudioRecordingController.cs (hunk 1 above)
Comment (valentinabennett) [reply VideoAPI/Video.API/Controllers/AudioRecordingController.cs:237]: renamed method

File: VideoAPI/Video.API/Controllers/ConferenceController.cs (hunk 3 above)
Comment (valentinabennett) [reply VideoAPI/Video.API/Controllers/ConferenceController.cs:403]: What about AI?

**Pull request merged**